scripts/check_docs_formatted.py

```diff
@@ -114,24 +111,66 @@
     "unexpected-indentation",
 ]
 
+SCRIPT_DIR = Path(__file__).resolve().parent
+ROOT_DIR = SCRIPT_DIR.parent
+
 
 class CodeBlockError(Exception):
     """A code block parse error."""
 
 
-def format_str(
-    src: str,
-    black_mode: black.FileMode,
-) -> tuple[str, Sequence[CodeBlockError]]:
-    """Format a single docs file string."""
+class InvalidInput(ValueError):
+    """Raised when ruff fails to parse file."""
+
+
+def format_str(code: str) -> str:
+    """Format a code block with ruff by writing to a temporary file."""
+    tmp_file = SCRIPT_DIR / "tmp.py"
+
+    # Create a temporary file to write the code block to
+    with tmp_file.open("w+") as f:
+        f.write(code)
+
+    # Run ruff to format the tmp file
+    try:
+        subprocess.run(
+            ["./target/debug/ruff", "format", tmp_file.as_posix()],
+            cwd=ROOT_DIR,
+            check=True,
+            capture_output=True,
+        )
```

Comment:
I think we should use stdin instead of a temp file. You can find an example here:

https://github.com/astral-sh/ruff/blob/5af48337a566b493a8263ff4c641462035a3c558/scripts/fuzz-parser/fuzz.py#L41-L46

This removes the need for handling temp files :)

Reply:
Updated in 11ef36c8d

Reply:
Is it possible to tell Ruff if the input code is from a `py` or `pyi` file? This will be useful to know for fixing https://github.com/astral-sh/ruff/issues/11568. If not, then a temp file approach may be required.